packages/sdk/src/utils.ts

```diff
@@ -184,15 +205,44 @@ export async function awaitResult<Result = unknown>(
 }
 
 /**
- * Enum encapsulating all the different types of targets used in the Boost V2 Protocol.
+ * Given a wagmi config and a map of chain id's to addresses, determine an address that maps to the currently connected chain id, or throw a typed error.
  *
  * @export
- * @enum {number}
+ * @param {Config} config
+ * @param {Record<string, Address>} addressByChainId
+ * @param {number} desiredChainId
+ * @returns {Address}
+ * @throws {@link NoConnectedChainIdError}
+ * @throws {@link InvalidProtocolChainIdError}
  */
-export enum RegistryType {
-  ACTION = 0,
-  ALLOW_LIST = 1,
-  BUDGET = 2,
-  INCENTIVE = 3,
-  VALIDATOR = 4,
+export function assertValidAddressByChainId(
```

Comment:
We use the wagmi `assertValidaAddress` in a bunch of places - we can prefer this without the `desiredChainId` going forward for flexibility/uniformity?

Reply:
`assertValidAddress` is mostly used for asserting that whatever method you're calling will attach a address to the underlying viem call. `assertValidAddressByChainId` is for selecting a base address given a chain id, useful for boost creation mainly